specs/driver_spec.rb

```diff
@@ -0,0 +1,159 @@
+require_relative 'spec_helper'
+
+describe "Driver" do
+  describe "constructor" do
+    before do
+      @name = 'Ada'
+      @id = 68
+      @vin = '123456789abcdefeg'
+    end
+
+    it "can be instantiated" do
+      driver = RideShare::Driver.new(name: @name, id: @id, vin: @vin)
+      driver.must_be_instance_of RideShare::Driver
+    end
+
+    it "takes a name, VIN, and ID" do
+      driver = RideShare::Driver.new(name: @name, id: @id, vin: @vin)
+
+      driver.must_respond_to :name
+      driver.name.must_equal @name
+
+      driver.must_respond_to :id
+      driver.id.must_equal @id
+
+      driver.must_respond_to :vin
+      driver.vin.must_equal @vin
+    end
+
+    it "raises an argument error if the VIN is not a string" do
+      proc {
+        RideShare::Driver.new(vin: 23456789064234576)
+      }.must_raise ArgumentError
+    end
+
+    it "raises an invalid VIN error if the VIN is not 17 characters long" do
+      invalid_vin = '123456dfsd'
+
+      proc {
+        RideShare::Driver.new(vin: invalid_vin)
+      }.must_raise RideShare::InvalidVinError
+    end
+
+    it "raises an invalid VIN error if the VIN has a character other than a letter or number" do
+      invalid_vin = '12345!!#2345ddegd'
+
+      proc {
+        RideShare::Driver.new(vin: invalid_vin)
+      }.must_raise RideShare::InvalidVinError
+    end
+  end
+
+  # As this method simply calls the Trip.find_all_for_driver method,
+  # extensive testing was not done.
+  describe "trips" do
+    it "returns only Trips that match this driver's id" do
+      driver_77 = RideShare::Driver.new(id: 77)
+      driver_77.trips.each do |trip|
+        trip.must_be_instance_of RideShare::Trip
+        trip.driver_id.must_equal driver_77.id
+      end
+    end
+
+    it "returns an empty array if driver id is undefined" do
```

Comment:
Good test case! Edge cases like this, asking what happens when there *isn't* any data, can be easy to miss.